src/main/java/uk/gov/gchq/palisade/resource/AbstractLeafResource.java

```diff
@@ -37,48 +36,76 @@ public AbstractLeafResource() {
     }
 
     public AbstractLeafResource type(final String type) {
-        requireNonNull(type, "The type of a resource cannot be set to null.");
-        this.type = type;
+        this.setType(type);
         return this;
     }
 
     public AbstractLeafResource serialisedFormat(final String serialisedFormat) {
-        requireNonNull(serialisedFormat, "The serialised format of a resource cannot be set to null.");
-        this.serialisedFormat = serialisedFormat;
+        this.setSerialisedFormat(serialisedFormat);
         return this;
     }
 
     public AbstractLeafResource attributes(final Map<String, Object> attributes) {
```

Comment:
This needs the @generated annotation on the mutator

Reply:
Done